backend/infrahub/core/query/relationship.py

```diff
@@ -581,23 +581,29 @@ async def query_init(self, db: InfrahubDatabase, **kwargs) -> None:  # pylint: d
         branch_level_str = "reduce(br_lvl = 0, r in relationships(path) | br_lvl + r.branch_level)"
         froms_str = db.render_list_comprehension(items="relationships(path)", item_name="from")
         query = """
-        MATCH (rl:Relationship { name: $rel_identifier })
+        MATCH (source_node:Node)%(arrow_left_start)s[:IS_RELATED]%(arrow_left_end)s(rl:Relationship { name: $rel_identifier })
+        WHERE source_node.uuid IN $source_ids AND $source_kind IN LABELS(source_node)
```

Comment:
Do we also need to filter on Labels since we are already filtering on IDS ? also scanning the labels is probably slower than scanning the IDS so removing labels might help with the performance.

Reply:
I don't know, I just took the WHERE clause that was inside the subquery. Your point is valid but I don't know if the `$source_kind IN LABELS(source_node)` is there to ensure `$source_ids` nodes are of a single kind. (if this is impossible, we could indeed remove this clause)

Reply:
Let's remove it before the subquery and we can leave it inside for now

Reply:
Done, thank you good catch :+1: